prov/usnic/src/usdf_fabric.c

```diff
@@ -75,12 +75,82 @@
 
 struct usdf_usnic_info *__usdf_devinfo;
 
+char *
+get_devname(uint32_t version, struct usd_device_attrs *dap)
+{
+	char *bufp;
+	struct in_addr in;
+	char *addrnetw;
+	size_t size;
+
+	if (FI_VERSION_GE(version, FI_VERSION(1, 4))) {
+		in.s_addr = dap->uda_ipaddr_be & dap->uda_netmask_be;
+		addrnetw = inet_ntoa(in);
+		size = snprintf(NULL, 0, "%s/%d", addrnetw, dap->uda_prefixlen) + 1;
+		bufp = calloc(1, size);
+		sprintf(bufp, "%s/%d", addrnetw, dap->uda_prefixlen);
+	} else {
+		bufp = strdup(dap->uda_devname);
+	}
+
+	return bufp;
+}
+
+enum check_devname_func_t {
+	CHECK_DEVNAME_GETINFO,
+	CHECK_DEVNAME_FABRIC
+};
+
+int
+check_devname(uint32_t version, struct usd_device_attrs *dap, struct fi_fabric_attr *fattrp,
+		enum check_devname_func_t caller)
+{
+	int ret;
+	char *devname;
+
+	char* devname_1_3 = get_devname(FI_VERSION(1,3), dap);
+	char* devname_1_4 = get_devname(FI_VERSION(1,4), dap);
```

Comment:
I talked it over with @piroux and decided that the functions should probably be named `check_fabric_name` and `get_fabric_name` since they aren't really tied to the device name at all.

Also we realized an insight here is that the version information will never be available in a call from `fi_fabric`, but will be available in a call from `fi_getinfo`. So if the version is available, we can just skip and check the version provided. Otherwise, we can check if there is a slash in the name and treat it appropriately. 

```c
bool check_fabric_name(uint32_t version, struct usd_device_attrs *dap, char *hint)
{
	bool valid = false;
	char *reference;

	USDF_DBG("checking devname: (version=%d) (devname='%s'\n", version,
			hint);

	if (version) {
		reference = get_fabric_name(version, dap);
		if (strmp(reference, hint) == 0)
			valid = true;

		free(reference);
		return valid;
	}

	/* The hint string itself is kind of a version check, in pre-1.4 the
	 * name was just 'usnic'. In 1.4 and beyond, then name is actually CIDR
	 * notation.
	 */
	if (strstr(hint, '/'))
		return check_fabric_name(FI_VERSION(1, 4), dap, hint);

	return check_fabric_name(FI_VERSION(1, 3), dap, hint);
}
```

I find this to be clearer, thoughts @jsquyres?